test/double_bucket_queue.cc

```diff
@@ -18,18 +19,28 @@ namespace {
 void TryAddRemove(const std::vector<uint32_t>& costs, const std::vector<uint32_t>& expectedorder) {
   std::vector<float> edgelabels;
 
-  const auto edgecost = [&edgelabels](const uint32_t label) { return edgelabels[label]; };
+  const auto edgecost = [&edgelabels](const uint32_t label) {
+    return edgelabels[label];
+  };
 
   uint32_t i = 0;
-  DoubleBucketQueue adjlist(0, 10000, 5, edgecost);
+  DoubleBucketQueue adjlist(0, 10000, 1, edgecost);
   for (auto cost : costs) {
     edgelabels.emplace_back(cost);
     adjlist.add(i);
     i++;
   }
   for (auto expected : expectedorder) {
+    // Do the same transform that's done in `edgecost()`
+    expected = (uint32_t)(float)expected;
     uint32_t labelindex = adjlist.pop();
-    EXPECT_EQ(edgelabels[labelindex], expected) << "TryAddRemove: expected order test failed";
+    auto edgelabel = 0;
+    if (labelindex != kInvalidLabel) {
+      edgelabel = edgelabels[labelindex];
+    } else {
+      fprintf(stderr, "#####################################  INVALID LABEL INDEX\n");
```

Comment:
Curious, why we're not using `cerr`?

Reply:
I found the format syntax so much simpler to use than the piping using `<<` operator, especially when trying to print in different representations. For this line in particular it didn't matter much I guess.

Reply:
It looks like the rest of the files in `test` use cout/cerr for console logs. I would prefer to avoid and 3rd style in the mix. If you feel strongly about adding `fprintf`, then thats fine too.